frontend/src/pages/pipelines/global/experiments/artifacts/useGetArtifactsList.ts

```diff
@@ -2,17 +2,47 @@ import React from 'react';
 
 import { usePipelinesAPI } from '~/concepts/pipelines/context';
 import { Artifact, GetArtifactsRequest } from '~/third_party/mlmd';
+import { ListOperationOptions } from '~/third_party/mlmd/generated/ml_metadata/proto/metadata_store_pb';
 import useFetchState, { FetchState } from '~/utilities/useFetchState';
+import { MlmdRequestProps } from '~/pages/pipelines/global/experiments/useMlmdRequestProps';
+
+export interface ArtifactsListResponse {
+  artifacts: Artifact.AsObject[];
+  nextPageToken: string;
+}
+
+const defaultMaxResultSize = 10;
 
 export const useGetArtifactsList = (
+  options?: Partial<
+    Pick<MlmdRequestProps, 'filterQuery' | 'maxResultSize' | 'pageToken' | 'orderBy'>
+  >,
   refreshRate?: number,
-): FetchState<Artifact.AsObject[] | null> => {
+): FetchState<ArtifactsListResponse | null> => {
   const { metadataStoreServiceClient } = usePipelinesAPI();
+  const { filterQuery, maxResultSize, pageToken } = options || {};
 
   const fetchArtifactsList = React.useCallback(async () => {
-    const response = await metadataStoreServiceClient.getArtifacts(new GetArtifactsRequest());
-    return response.toObject().artifactsList;
-  }, [metadataStoreServiceClient]);
+    const request = new GetArtifactsRequest();
+    const listOperationOptions = new ListOperationOptions();
+
+    if (filterQuery) {
+      listOperationOptions.setFilterQuery(filterQuery);
+    }
+
+    if (pageToken) {
+      listOperationOptions.setNextPageToken(pageToken);
+    }
+
+    listOperationOptions.setMaxResultSize(maxResultSize ?? defaultMaxResultSize);
+    request.setOptions(listOperationOptions);
+
+    const response = await metadataStoreServiceClient.getArtifacts(request);
+    const nextPageToken = response.getNextPageToken();
+    listOperationOptions.setNextPageToken(nextPageToken);
+
+    return { artifacts: response.toObject().artifactsList, nextPageToken };
```

Comment:
You can use `response.getArtifactsList()` instead of the conversion by yourself.

Reply:
They're not the same. That's a different set of data.

Reply:
Wow, what's the difference because I am using `response.getExecutionList()` to get the executions

Reply:
Yes, because I think the intention of the KubeFlow MLMD class is to let you use the method in the class, though it's not documented anywhere. So if you are using `response.getArtifactsList()`, then `artifact.getCreateTimeSinceEpoch()` should give you the same result compared to this line https://github.com/opendatahub-io/odh-dashboard/pull/2744/files#diff-a85ef6d786c1a8509e52597a67c0273d4302a82018ad211b237f1d45bb1f82baR180

Reply:
Are you claiming the theoretical intention of the MLMD class and the accessing of values via method call is better in some way than just accessing from the object keys?

Reply:
Yes, because this is how they do in the codebase. Also, it matches the type `Artifact` instead of type `Artifact.AsObject`

Reply:
How is it better? They expose AsObject for a reason. Just because they're not using it in their code is not justification for it being better. Unless there's a good reason, I'll be leaving as-is. The consumption of data in the table is more straightforward this way.

Reply:
MLMD generated functions exposes it, not KF. We should use the full functionality of the mlmd resource types which include more than just the data, there are other functions we would not have if we convert the artifact into an object right away. We may not use them know, but maybe in the future we will. I would suggest we keep the main hooks returning the base type, and then further down into implementation, we we can choose to convert that artifact into an object.

Reply:
Why would we want anything other than the data here? Is that not what every other API fetching hook we have does?

Reply:
I am trying to keep consistency across the feature. If we are using one type everywhere else, but then another version of the type in one other spot, it can get confusing. And you can still use that object further down in implementation.

Reply:
Where would we use Artifact as a non-object form and why?

Reply:
I would like to know what the answers to these questions are, but since this seems to be blocking the PR, I went ahead and updated my hook to return the list in a non-object form, and am just having to iterate over all artifacts now for the table, which is more code than what was there prior.

Reply:
We are not using it now, but for consistency across all of our other MLMD hooks